assets/src/js/admin/stripe-admin.js

```diff
@@ -232,6 +233,138 @@ window.addEventListener( 'DOMContentLoaded', function() {
+                    inputField.style.marginBottom = '10px';
+                    saveButton.style.marginRight = '5px';
+
+                    // Add events.
+                    inputField.addEventListener(
+                        'keyup',
+                        (e) => {
+                            let newStatementDescriptor = inputField.value.trim();
+
+                            if (!newStatementDescriptor || newStatementDescriptor === getStripeStatementDescriptorText()) {
+                                saveButton.disabled = true;
+                                return;
+                            }
+
+                            saveButton.disabled = false;
+                        }
+                    );
+
+                    cancelButton.addEventListener(
+                        'click',
+                        (e) => {
+                            e.preventDefault();
+                            exitStatementDescriptorEditingMode()
+                        });
+
+                    saveButton.addEventListener(
+                        'click',
+                        (e) => {
+                            e.preventDefault();
+
+                            let newStatementDescriptorText = getNewStatementDescriptor();
+                            let actionUrl = `${container.getAttribute('data-action-url')}&statement-descriptor=${encodeURIComponent(newStatementDescriptorText)}`
+
+                            if( ! isValidaStatementDescriptor(newStatementDescriptorText) ) {
+                                new Give.modal.GiveErrorAlert({
+                                    modalContent:{
+                                        title: __( 'Invalid Statement Descriptor Text', 'give'),
+                                        desc: sprintf(
+                                            '%s <br><a href="%s" target="_blank">%s</a>',
+                                            __( 'Please enter a valid Stripe statement descriptor..', 'give'),
+                                            'https://stripe.com/docs/statement-descriptors#requirements',
+                                            __( 'Read more about stripe statement descriptor text requirements.', 'give'),
+                                        ),
+                                    }
+                                }).render();
+                                return;
+                            }
+
+                            fetch(actionUrl)
+                                .then(response => response.json())
+                                .then(response => {
+                                    if( ! response.success ){
+                                        new Give.modal.GiveErrorAlert({
+                                            modalContent:{
+                                                title: __( 'Unable To Update Stripe Statement Descriptor', 'give'),
+                                                desc: sprintf(
```

Comment:
@Benunc Can you provide feedback on this error message? This message will appear in a modal.

Reply:
Same with this error @ravinderk. If it's showing in a modal, we need to give the user a next step.